src/go/rpk/pkg/system/cpu.go

```diff
@@ -9,8 +9,60 @@
 
 package system
 
-import "github.com/shirou/gopsutil/v3/cpu"
+import (
+	"errors"
+	"strconv"
+	"strings"
 
-func CpuInfo() ([]cpu.InfoStat, error) {
-	return cpu.Info()
+	"github.com/spf13/afero"
+)
+
+type CPUInfo struct {
+	ModelName	string
+	Cores		int
+}
+
+func CpuInfo(fs afero.Fs) ([]*CPUInfo, error) {
+	bytes, err := afero.ReadFile(fs, "/proc/cpuinfo")
+	if err != nil {
+		return nil, err
+	}
+	contents := string(bytes)
+	if contents == "" {
+		return nil, errors.New("/proc/cpuinfo is empty")
+	}
+	sections := strings.Split(contents, "\n\n")
+
+	cpus := []*CPUInfo{}
+	for _, s := range sections {
+		if s == "" {
+			continue
+		}
+		cpu := &CPUInfo{Cores: 1}
+		sectionLines := strings.Split(s, "\n")
+		for _, l := range sectionLines {
+			row := strings.Split(l, ":")
+			for i := 0; i < len(row); i++ {
+				row[i] = strings.TrimSpace(row[i])
+			}
+			if len(row) < 2 {
+				continue
+			}
+			k := row[0]
+			v := row[1]
+
+			switch k {
+			case "model name":
+				cpu.ModelName = v
+			case "cpu cores":
+				cores, err := strconv.Atoi(v)
```

Comment:
I think a different order, would yield partial information.. 

Why not do: 

1. Trim N-1 cpus. 
2. Fill up the struct - don't early return
3. return filled in struct. 


Also, note some existing go-only libs:

https://github.com/c9s/goprocinfo/blob/master/linux/cpuinfo.go#L70-L133

https://github.com/prometheus/procfs/blob/master/cpuinfo.go

Reply:
so my main comment here is not to go on and tear this out, but that the return below will yield partial information.

Reply:
actually a better strategy is this: 

1. start at the top of the file. 
2. start a struct
3. if struct is filled in, return
4. prase all fields
5. in case of error, return nil + error

Reply:
Per our discussion, we'll keep the slice return type and return `nil, err` in case of error.